lib/vm/src/module.rs

```diff
@@ -115,6 +118,142 @@ pub struct ModuleInfo {
     pub num_imported_globals: usize,
 }
 
+/// Mirror version of ModuleInfo that can derive rkyv traits
+#[derive(RkyvSerialize, RkyvDeserialize, Archive)]
+pub struct RkyvModuleInfo {
```

Comment:
`ArchivedModuleInfo` seems like a better name?

Reply:
`derive(Archive)` will generate a `ArchivedModuleInfo` so i can't use that name. Also this mean to be a before-archive ModuleInfo-equivalant struct, maybe "RkyvArchivableModuleInfo" or "ModuleInfoForArchive"?

Reply:
But the `ModuleInfo` struct is not deriving from `Archive`, hence `ArchivedModuleInfo ` should be available.
So I guess it will be fine?

Reply:
Ah yes, it should be available. however this shouldn't be named so, because it hasn't been archived.
ArchivableModuleInfo
RkyvArchivableModuleInfo
ModuleInfoForArchive
or ModuleInfoBeforeArchive?

Note derive(archive) would prefix Archived to this name, and we're likely need to refer this ArchivedXXX name, therefore (Archived)ArchivableModuleInfo maybe the best?

Reply:
Let's go with `ArchivableModuleInfo` 👍